src/plugins/drawing-tool/components/drawing-tool.tsx

```diff
@@ -37,9 +38,34 @@ const DrawingToolComponent: React.FC<IProps> = (props) => {
         return getTitle();
       }
     });
-
+    hotKeys.current.register({
+      "cmd-v": handlePaste, //allows user to paste image with cmd+v
+    });
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
+  const handlePaste = () => {
+    pasteImage();
+  };
+
+  const pasteImage = async()=>{
+    const clipboardContents = await navigator.clipboard.read();
+    if (clipboardContents.length > 0){
+      if (clipboardContents[0].types.includes("image/png")){
+        clipboardContents[0].getType("image/png").then(blob=>{
+          const blobToFile = new File([blob], "clipboard-image");
+          uploadImage(blobToFile);
+        });
+      }
+    }
+  };
+
+  const uploadImage = (file: File) => {
+    gImageMap.addFileImage(file)
+      .then(image => {
+        setImageUrlToAdd(image.contentUrl || '');
+      });
+  };
```

Comment:
Rather than duplicating this code in every tile we should share what we can. For instance, a new `clipboard-utils.ts` module could contain a function like:
```typescript
  const pasteClipboardImage = async (onComplete: (imageUrl: string) => void) => {
    const clipboardContents = await navigator.clipboard.read();
    if (clipboardContents.length > 0) {
      if (clipboardContents[0].types.includes("image/png")) {
        clipboardContents[0].getType("image/png").then(blob => {
          const blobToFile = new File([blob], "clipboard-image");
          const image = await gImageMap.addFileImage(file);
          onComplete(image.contentUrl || '');
        });
      }
    }
  };
```
and then the calling code would be something like:
```typescript
  const handlePaste = () => {
    pasteClipboardImage(imageUrl => setImageUrlToAdd(imageUrl));
  };
```

Reply:
Makes sense, will refactor today.